src/gsi/setuprad.f90

```diff
@@ -996,6 +999,20 @@ subroutine setuprad(obsLL,odiagLL,lunin,mype,aivals,stats,nchanl,nreal,nobs,&
            varinv(1:nchanl) = zero
         endif
 
+! Include a separate check for NaNs in CRTM calculations.  These are not always caught by other tests
+
+        if (ANY(ieee_is_nan(tsim(1:nchanl)))) then
+           write(*,*) 'WARNING: NaN found in CRTM simulated radiance output'
```

Comment:
Suggestion: add `trim(isis)` to the write statement so we know what sensor_satellite is producing a NaN.    Same comment applies to PR #925.